frontend/src/pages/modelRegistry/screens/RegisterModel/RegisterModel.tsx

```diff
@@ -36,12 +42,17 @@ const RegisterModel: React.FC = () => {
   const [isSubmitting, setIsSubmitting] = React.useState(false);
   const [submitError, setSubmitError] = React.useState<Error | undefined>(undefined);
   const [formData, setData] = useRegisterModelData();
-  const isModelNameValid = isNameValid(formData.modelName);
-  const isSubmitDisabled = isSubmitting || isRegisterModelSubmitDisabled(formData);
-  const { modelName, modelDescription } = formData;
   const [registeredModelName, setRegisteredModelName] = React.useState<string>('');
   const [versionName, setVersionName] = React.useState<string>('');
   const [errorName, setErrorName] = React.useState<string | undefined>(undefined);
+  const [registeredModels, registeredModelsLoaded, registeredModelsLoadError] =
+    useRegisteredModels();
+
+  const isModelNameValid = isNameValid(formData.modelName);
```

Comment:
It looks like with the refactor, the submit disabled logic is working but the new error never renders. I think it's because `isModelNameValid` is what we're blocking rendering the error on and now you've split the duplicate check out of it.

Maybe we need a new `const shouldShowError = !isModelNameValid || isModelNameDuplicate` and use that for the `validated` prop and the conditional for the error below.